src/Frontend/Document.php

```diff
@@ -158,6 +163,16 @@ public function render(): string
         return $this->makeView()->render();
     }
 
+    /**
+     * Prepares the document to be populated.
+     *
+     * @return void
+     */
+    public function prepare()
+    {
+        $this->premadeData['title'] = $this->makeTitle();
```

Comment:
I don't like this. 

```php
        $this->premadeData['title'] = $this->makeTitle();
```

`makeTitle` assigns the value of the array itself. Why can't we do something like:

```suggestion
        $this->premadeData['title'] = $this->premadeData['title'] ?? $this->makeTitle();
```

And then remove this in makeTitle:

```
```php
        if (isset($this->premadeData['title'])) {
            return $this->premadeData['title'];
        }

```